src/args/arg_matches.rs

```diff
@@ -1,282 +1,572 @@
-    /// if let Some(os) = matches.values_of("output") {
-    ///        for o in os {
-    ///            println!("A value for output: {}", o);
-    ///        }
-    /// }
+    /// use std::ffi::OsString;
+    /// use std::os::unix::ffi::OsStrExt;
+    ///
+    /// let m = App::new("utf8")
+    ///     .arg(Arg::from_usage("<arg> 'some arg'"))
+    ///     .get_matches_from(vec![OsString::from("myprog"),
+    ///                             // "Hi {0xe9}!"
+    ///                             OsString::from_vec(vec![b'H', b'i', b' ', 0xe9, b'!'])]);
+    /// assert_eq!(&*m.lossy_value_of("arg").unwrap(), "Hi \u{FFFD}!");
     /// ```
-    pub fn values_of(&'a self, name: &str) -> Option<Vec<&'a str>> {
-        if let Some(ref arg) = self.args.get(name) {
-            if let Some(ref vals) = arg.values {
-                return Some(vals.values().map(|s| &s[..]).collect::<Vec<_>>());
+    pub fn lossy_value_of<S: AsRef<str>>(&'a self, name: S) -> Option<Cow<'a, str>> {
+        if let Some(arg) = self.args.get(name.as_ref()) {
+            if let Some(v) = arg.vals.values().nth(0) {
+                return Some(v.to_string_lossy());
             }
         }
         None
     }
 
-    /// Returns if an argument was present at runtime.
+    /// Gets the OS version of a string value of a specific argument. If the option wasn't present
+    /// at runtime it returns `None`. An OS value on Unix-like systems is any series of bytes,
+    /// regardless of whether or not they contain valid UTF-8 code points. Since `String`s in Rust
+    /// are garunteed to be valid UTF-8, a valid filename on a Unix system as an argument value may
+    /// contain invalid UTF-8 code points.
     ///
+    /// *NOTE:* If getting a value for an option or positional argument that allows multiples,
+    /// prefer `os_values_of()` as `os_value_of()` will only return the *first* value.
     ///
     /// # Examples
     ///
-    /// ```no_run
+    /// ```ignore
     /// # use clap::{App, Arg};
-    /// # let matches = App::new("myapp")
-    /// #     .arg(Arg::with_name("output").takes_value(true)).get_matches();
-    /// if matches.is_present("output") {
-    ///        println!("The output argument was used!");
-    /// }
+    /// use std::ffi::OsString;
+    /// use std::os::unix::ffi::OsStrExt;
+    ///
+    /// let m = App::new("utf8")
+    ///     .arg(Arg::from_usage("<arg> 'some arg'"))
+    ///     .get_matches_from(vec![OsString::from("myprog"),
+    ///                             // "Hi {0xe9}!"
+    ///                             OsString::from_vec(vec![b'H', b'i', b' ', 0xe9, b'!'])]);
+    /// assert_eq!(&*m.os_value_of("arg").unwrap().as_bytes(), [b'H', b'i', b' ', 0xe9, b'!']);
     /// ```
-    pub fn is_present(&self, name: &str) -> bool {
-        if let Some(ref sc) = self.subcommand {
-            if sc.name == name {
-                return true;
-            }
+    pub fn os_value_of<S: AsRef<str>>(&self, name: S) -> Option<&OsStr> {
```

Comment:
I am not sure about the naming. The rust std library uses `args_os`, which puts the `os` part at the end. `Clap` also have used `get_matches_*`, which puts the method variants at the end as well. Same with `lossy_value_of`.